frontend/catalyst/api_extensions/callbacks.py

```diff
@@ -271,7 +358,10 @@ def callback_implementation(
 
     results_aval = tree_map(convert_pytype_to_shaped_array, result_shape_dtypes)
     flat_results_aval, out_tree = tree_flatten(results_aval)
-    memref_callable = MemrefCallable(cb, results_aval, *args, **kwargs)
+    if device is None:
+        memref_callable = MemrefCallable(cb, results_aval, *args, **kwargs)
+    else:
+        memref_callable = JaxJitCallable(cb, device, results_aval, *args, **kwargs)
```

Comment:
A final question, also for @josh146, but did we want to enforce that `accelerate` actually runs functions on an accelerator? That is, we still can take the default jax device but if it is the cpu (meaning no accelerator was detected), we would raise an error.

Reply:
@dime10 for now no I would like to keep it working with `jax.jit` and CPUs, as I see that being super useful.

We briefly discussed having a separate `catalyst.gpu` function that simply adds in this validation, which may make sense.

Reply:
I thought the argument against `gpu` was that it was too specific (that is there are types of accelerators), hence `accelerate`. Fairly certain no-one would consider a CPU an accelerator. 

But keeping it working on CPU is fine by me, it may just mean users don't actually run on gpu when they expect (if for some reason it is not detected).

Reply:
- We could query what is the default backend every time we use `accelerate`. If it says `gpu` or `tpu` we ignore it, cause it means we have at least one accelerator device. But if it says `cpu`, we prompt a warning to the user, so they know they are not actually using an accelerator:
```
import jax
if jax.default_backend() == 'cpu':
    # warning
```

- Should we create a wrapper that lists all the available accelerators?
```
jax.devices('gpu')
jax.devices('tpu')
```

Reply:
A warning could work, although I know @josh146 is not fond of warnings if they cannot be fixed by the user. If the `accelerate` decorator is explicitly meant to be useable on CPU (in what sense are we accelerating in this case?) then I think a warning would not fit that criteria.

Reply:
These are all good points 🤔 

I'm in favour of leaving it without an error for now, and allowing it to work with `jax.jit` and CPUs (because I imagine I will be helping people with code and we'll want it to work on CPUs for testing).

There are a couple of reasons I think this functionality will be super useful:

- Unlike `pure_callback`, `accelerate` allows you to use `jax.jit` functions _without needing to manually specify the return type_, which could be quite complex for a new user!

- In the future, if we can interface with `jax.jit` functions _without a Python callback in the runtime_, then I see `accelerate(jax_jit_func)(*args)` providing value even if run on CPUs. At this point, maybe we break the function into two (with two different names) --- one that raises a warning for non-accelerators, and one that doesn't.

Reply:
Ok, if I understand correctly, we are leaving this as is and in the future maybe raise a warning. I'll resolve this conversation then.